source/distributed/fully_distributed_tria.cc

```diff
@@ -407,29 +407,297 @@ namespace parallel
+        ierr = MPI_File_write_at(fh,
+                                 myrank * sizeof(unsigned int),
+                                 DEAL_II_MPI_CONST_CAST(&buffer_size),
+                                 1,
+                                 MPI_UNSIGNED,
+                                 MPI_STATUS_IGNORE);
+        AssertThrowMPI(ierr);
+
+        // Write buffers to file.
+        ierr = MPI_File_write_at(fh,
+                                 mpisize * sizeof(unsigned int) +
+                                   offset, // global position in file
+                                 DEAL_II_MPI_CONST_CAST(buffer.data()),
+                                 buffer.size(), // local buffer
+                                 MPI_CHAR,
+                                 MPI_STATUS_IGNORE);
+        AssertThrowMPI(ierr);
+
+        ierr = MPI_File_close(&fh);
+        AssertThrowMPI(ierr);
+      }
+#else
       (void)filename;
 
-      AssertThrow(false, ExcNotImplemented());
+      AssertThrow(false, ExcNeedsMPI());
+#endif
     }
 
 
 
     template <int dim, int spacedim>
     void
     Triangulation<dim, spacedim>::load(const std::string &filename,
-                                       const bool         autopartition)
+                                       const bool /*autopartition*/)
     {
+#ifdef DEAL_II_WITH_MPI
+      Assert(this->n_cells() == 0,
+             ExcMessage("load() only works if the Triangulation is empty!"));
+
+      // Compute global offset for each rank.
+      unsigned int n_locally_owned_cells = this->n_locally_owned_active_cells();
+
+      unsigned int global_first_cell = 0;
+
+      int ierr = MPI_Exscan(&n_locally_owned_cells,
+                            &global_first_cell,
+                            1,
+                            MPI_UNSIGNED,
+                            MPI_SUM,
+                            this->mpi_communicator);
+      AssertThrowMPI(ierr);
+
+      global_first_cell *= sizeof(unsigned int);
+
+
+      unsigned int version, numcpus, attached_count_fixed,
+        attached_count_variable, n_global_active_cells;
+      {
+        std::string   fname = std::string(filename) + ".info";
+        std::ifstream f(fname.c_str());
+        AssertThrow(f, ExcIO());
+        std::string firstline;
+        getline(f, firstline); // skip first line
+        f >> version >> numcpus >> attached_count_fixed >>
+          attached_count_variable >> n_global_active_cells;
+      }
+
+      AssertThrow(version == 4,
+                  ExcMessage("Incompatible version found in .info file."));
+      Assert(this->n_global_active_cells() == n_global_active_cells,
+             ExcMessage("Number of global active cells differ!"));
+
+      // Load description and construct the triangulation.
+      {
+        const int myrank =
+          Utilities::MPI::this_mpi_process(this->mpi_communicator);
+        const int mpisize =
+          Utilities::MPI::n_mpi_processes(this->mpi_communicator);
+
```

Comment:
To be on the safe side:
```suggestion
        const int mpisize =
          Utilities::MPI::n_mpi_processes(this->mpi_communicator);
        AssertDimension(numcpus, mpisize);
```

Maybe you could also add to the documentation of `p:f:T::load()` a description of the limitations of this function.